src/libs/ValidationUtils.ts

```diff
@@ -20,16 +20,24 @@ import StringUtils from './StringUtils';
  */
 function validateCardNumber(value: string): boolean {
     let sum = 0;
-    for (let i = 0; i < value.length; i++) {
-        let intVal = parseInt(value.substr(i, 1), 10);
-        if (i % 2 === 0) {
+    let shouldDouble = false;
+
+    // Loop through the card number from right to left
+    for (let i = value.length - 1; i >= 0; i--) {
+        let intVal = parseInt(value[i], 10);
+
+        // Double every second digit from the right
+        if (shouldDouble) {
```

Comment:
NAB, but maybe there is some better explanation we could add here about the doubling stuff?